NitroxPatcher/Patches/Persistent/GameSettings_SerializeInputSettings_Patch.cs

```diff
@@ -22,11 +24,29 @@ public static void Postfix(GameSettings.ISerializer serializer)
                 {
                     Log.Debug($"Getting keybinding: {keyBinding.Device}, {keyBinding.Label} ({keyBinding.Button}), {bindingSet}");
                     string binding = GameInput.GetBinding(keyBinding.Device, keyBinding.Button, bindingSet);
-                    string name = string.Format(serializerFormat, keyBinding.Device, keyBinding.Button, bindingSet);
 
-                    GameInput.SetBinding(keyBinding.Device, keyBinding.Button, bindingSet, serializer.Serialize(name, binding));
+                    // We need to assign the correct binding for primary and secondary binding sets to the relevant area of the config.
+                    switch (keyBinding.Button)
+                    {
+                        case (GameInput.Button)KeyBindingValues.CHAT when bindingSet == GameInput.BindingSet.Primary:
+                            cfg.OpenChatKeybindPrimary = binding;
+                            break;
+                        case (GameInput.Button)KeyBindingValues.FOCUS_DISCORD when bindingSet == GameInput.BindingSet.Primary:
+                            cfg.FocusDiscordKeybindPrimary = binding;
+                            break;
+                        case (GameInput.Button)KeyBindingValues.CHAT when bindingSet == GameInput.BindingSet.Secondary:
+                            cfg.OpenChatKeybindSecondary = binding;
+                            break;
+                        case (GameInput.Button)KeyBindingValues.FOCUS_DISCORD when bindingSet == GameInput.BindingSet.Secondary:
+                            cfg.FocusDiscordKeybindSecondary = binding;
+                            break;
+                    }
```

Comment:
Code can be a bit simplified:

```suggestion
                    switch ((KeyBindingValues)keyBinding.Button)
                    {
                        case KeyBindingValues.CHAT when bindingSet == GameInput.BindingSet.Primary:
                            cfg.OpenChatKeybindPrimary = binding;
                            break;
                        case KeyBindingValues.FOCUS_DISCORD when bindingSet == GameInput.BindingSet.Primary:
                            cfg.FocusDiscordKeybindPrimary = binding;
                            break;
                        case KeyBindingValues.CHAT when bindingSet == GameInput.BindingSet.Secondary:
                            cfg.OpenChatKeybindSecondary = binding;
                            break;
                        case KeyBindingValues.FOCUS_DISCORD when bindingSet == GameInput.BindingSet.Secondary:
                            cfg.FocusDiscordKeybindSecondary = binding;
                            break;
                    }
```